### ISSUE TITLE
Build Instructions

### ISSUE THREAD

**Issue by krishpranav** [PureDarwin/PureDarwin#82]
hey @PureDarwin team can you provide a clean build instructions for building PureDarwin

**Comment by wjk** [comment]
Greetings! I'm actually working on documenting that right now. To build PureDarwin, you'll need to do the following:

1. Be on a Mac with Xcode 12 or later installed. This is because PureDarwin uses one tool (`iig`) from Xcode that is not open-source. In my spare time I am working on rewriting `iig`, which if successful would enable building PureDarwin on Linux as well.
2. Install CMake 3.15.1 or later, as well as the Ninja build tool. (e.g. `brew install cmake ninja`)
2. Clone the repo. Enter a directory called `build` in the root of the repository.
3. Run `cmake -G Ninja ..` from within the build directory. Once this succeeds, run `ninja`.

At this point, if there no errors were emitted by CMake or by Ninja, PureDarwin has been completely built. Unfortunately, this doesn't mean all that much right now, as PD is currently nowhere near being able to build a bootable image. We are currently working on integrating the libSystem component libraries into the repo, which must be completed before we will be able to link any user-mode binaries. (The other repositories under the PureDarwin organization on GitHub are legacy artifacts from a prior attempt at creating a build system for PureDarwin; their contents will be integrated into this repo when possible. The goal is for the PureDarwin repository to contain a complete, standalone copy of the operating system that can be compiled with only the standard Unix command-line tools and clang.)

Contributions are of course welcome, but PureDarwin is a very complicated codebase, and you will need detailed knowledge of the low-level system libraries to be able to write functional code for this project at this time. If you feel like you can contribute, please open a PR and I'll give you guidance there. Thanks!

**Comment by krishpranav** [comment]
hey @wjk thanks for the steps that you provide I will try these

**Comment by krishpranav** [comment]
I am on macOS big sur I am facing this issue
```
CMake Error at /usr/local/Cellar/cmake/3.21.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.21.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.21.1/share/cmake/Modules/FindOpenSSL.cmake:574 (find_package_handle_standard_args)
  tools/cctools/ld64/src/ld/CMakeLists.txt:4 (find_package)
```

**Comment by csekel** [comment]
Seems like a cmake issue, see if you can install openssl via brew .

**Comment by krishpranav** [comment]
@csekel I've installed openssl and I have added openssl path in .zshrc 
but still I am facing the same issue

**Comment by krishpranav** [comment]
https://stackoverflow.com/questions/16248775/cmake-not-able-to-find-openssl-library
I saw the solution in that they told to set openssl path in default $SHELL so I've added it to my .zshrc 
But still facing the same issue 😢

**Comment by wjk** [comment]
@krishpranav I have installed OpenSSL via Homebrew on my dev box, and CMake finds it automatically without any problems. But seeing as yours for some reason does not, you'll need to add the argument `-DOPENSSL_ROOT_DIR=/usr/local/opt/openssl` to your CMake command line. Hope this helps!

**Comment by krishpranav** [comment]
ok let me check it out

**Comment by krishpranav** [comment]
it is now building

**Comment by krishpranav** [comment]
but the problem now is If I run ninja run it is not running

**Comment by krishpranav** [comment]
➜  build git:(master) ✗ ninja run
ninja: error: unknown target 'run', did you mean 'xnu'?

**Comment by wjk** [comment]
As I said above, you can't run PureDarwin yet. The error that ninja is giving you is because you are telling it to build a target named `run`, which doesn't exist. I have never seen a CMake-based project that can be run through this method. (Normally, if you want to run a CMake-generated binary, you run it directly from the directory into which the linker puts it.)

**Comment by krishpranav** [comment]
ok thanks for clearing my doubts @wjk & @PureDarwin team

**Comment by hideout** [comment]
enjoy.

Am So., 8. Aug. 2021 um 05:04 Uhr schrieb Krisna Pranav <
***@***.***>:

**Comment by krishpranav** [comment]
hi @PureDarwin team I am facing a new issue so can you help me to fix it?

**Comment by krishpranav** [comment]
```
CMake Error at CMakeLists.txt:9 (message):
  You must specify one of KERNEL_BUILD_FREEBSD or KERNEL_BUILD_XNU
```

**Comment by hideout** [comment]
hey krisna,

if you read the error you should know what todo  ;)

 KERNEL_BUILD_XNU

—


cheers Ferdinand


Am Do., 2. Sept. 2021 um 09:22 Uhr schrieb Krisna Pranav <
***@***.***>:

**Comment by krishpranav** [comment]
I don't have any idea on that error
